File: src/6_hello_service/main_pi.cpp
// Simple program to blink LED in C++ on Pi

// Includes
#include <iostream>
#include <wiringPi.h>

using namespace std;

// Defines
#define BLINK_LED 8
#define BLINK_DELAY 500

// Runtime entrypoint
int main()
{
    // Initialize Wiring Pi
    cout << "Initializing Pi GP I/O" << endl;
    if(wiringPiSetup()!=0)
    {
        cout << "Wiring Pi Setup Failed" << endl;
    }

    // Initializing output
    cout << "Running LED blink" << endl;
    pinMode(BLINK_LED, OUTPUT);
    digitalWrite(BLINK_LED, LOW);

    // Main loop; continuously blink LED
    while(1)
    {
        // LED ON
        digitalWrite(BLINK_LED, HIGH);
        delay(BLINK_DELAY);
        // LED OFF
        digitalWrite(BLINK_LED, LOW);
        delay(BLINK_DELAY);
    }
}

File: src/5_hello_pipy/main_pi.py
# Simple program to blink LED in python on Pi

# Imports
import wiringpi as wp

# Defines
BLINK_LED = 8
BLINK_DELAY = 500

# Runtime entrypoint
def main():
    # Initialize Wiring Pi
    print("Initializing Pi GP I/O")
    if wp.wiringPiSetup()!=0:
        print("Wiring Pi Setup Failed")

    # Initializing output
    print("Running LED blink")
    wp.pinMode(BLINK_LED, wp.OUTPUT)
    wp.digitalWrite(BLINK_LED, wp.LOW)

    # Main loop; continuously blink LED
    while True:
        # LED ON
        wp.digitalWrite(BLINK_LED, wp.HIGH)
        wp.delay(BLINK_DELAY)
        # LED OFF
        wp.digitalWrite(BLINK_LED, wp.LOW)
        wp.delay(BLINK_DELAY)

# Call main
main()
